### PR TITLE
Fixing multiple protocol issues (May 28th, 2021)

### PR DESCRIPTION
**Pull request by scorpion9979**
Addresses the following issues: https://github.com/hifi-finance/hifi-protocol/issues/47, https://github.com/hifi-finance/hifi-protocol/issues/46, https://github.com/hifi-finance/hifi-protocol/issues/28, ..

### CODE REVIEW

File: helpers/numbers.ts [hunk 1]
```diff
@@ -18,3 +18,9 @@ export function price(x: string): BigNumber {
 export function usdc(x: string): BigNumber {
   return fp(x, 6);
 }
+
+export function tokenWithNDecimalsPrecisionScalar(n: number): BigNumber {
+  if (n > 18 || n < 0) throw new Error(`Invalid n given: ${n}`);
```
Comment (paulrberg) [inline helpers/numbers.ts:23]: Can we add brackets here so that the styling remains consistent across the board?

File: helpers/numbers.ts [hunk 2]
```diff
@@ -18,3 +18,9 @@ export function price(x: string): BigNumber {
...
+  if (n === 18) return bn("1");
```
Comment (paulrberg) [inline helpers/numbers.ts:24]: Ditto.

File: test/unit/chainlinkOperator/ChainlinkOperator.behavior.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-import shouldBehaveLikeGetAdjustedPrice from "./view/getAdjustedPrice";
+import shouldBehaveLikegetNormalizedPrice from "./view/getNormalizedPrice";
 import shouldBehaveLikeGetFeed from "./view/getFeed";
 import shouldBehaveLikeGetPrice from "./view/getPrice";
 
 export function shouldBehaveLikeChainlinkOperator(): void {
   describe("View Functions", function () {
-    describe("getAdjustedPrice", function () {
-      shouldBehaveLikeGetAdjustedPrice();
+    describe("getNormalizedPrice", function () {
```
Comment (paulrberg) [inline test/unit/chainlinkOperator/ChainlinkOperator.behavior.ts:9]: Can we arrange these describe blocks alphabetically?

**Review by paulrberg — CHANGES_REQUESTED** [review]

File: contracts/oracles/ChainlinkOperator.sol [hunk 4]
```diff
@@ -31,10 +31,10 @@ contract ChainlinkOperator is
     /// CONSTANT FUNCTIONS ///
 
     /// @inheritdoc IChainlinkOperator
-    function getAdjustedPrice(string memory symbol) external view override returns (uint256) {
+    function getNormalizedPrice(string memory symbol) external view override returns (uint256) {
```
[the remark this answers is not in the document]
Comment (scorpion9979) [reply contracts/oracles/ChainlinkOperator.sol]: done

File: contracts/oracles/IChainlinkOperator.sol [hunk 5]
```diff
@@ -67,7 +67,7 @@ interface IChainlinkOperator {
     ///
     /// @param symbol The Erc20 symbol of the token for which to query the price.
     /// @return The normalized price.
-    function getAdjustedPrice(string memory symbol) external view returns (uint256);
+    function getNormalizedPrice(string memory symbol) external view returns (uint256);
```
[the remark this answers is not in the document]
Comment (scorpion9979) [reply contracts/oracles/IChainlinkOperator.sol]: done

File: helpers/constants.ts [hunk 6]
```diff
@@ -19,14 +19,16 @@ export const GAS_LIMIT_HARDHAT: BigNumber = bn("1e8");
...
 export const H_TOKEN_DECIMALS: BigNumber = bn("18");
 export const H_TOKEN_EXPIRATION_TIME: BigNumber = now().add(daysInSeconds(90));
-export const H_TOKEN_NAME: string = "Hifi USDC (2022-03-31)";
-export const H_TOKEN_SYMBOL: string = "hUSDC22Jun";
+export const H_TOKEN_EXPIRATION_DATE = new Date(H_TOKEN_EXPIRATION_TIME.toNumber() * 1000);
+export const H_TOKEN_EXPIRATION_DATE_FULL = H_TOKEN_EXPIRATION_DATE.toISOString().slice(0, 10);
+export const H_TOKEN_EXPIRATION_DATE_SHORT =
+  H_TOKEN_EXPIRATION_DATE.toLocaleString("default", { year: "2-digit" }).toUpperCase() +
+  H_TOKEN_EXPIRATION_DATE.toLocaleString("default", { month: "short" }).toUpperCase();
+export const H_TOKEN_NAME: string = `Hifi USDC (${H_TOKEN_EXPIRATION_DATE_FULL})`;
```
[the remark this answers is not in the document]
Comment (scorpion9979) [reply helpers/constants.ts]: done

File: helpers/constants.ts [hunk 7]
```diff
@@ -19,14 +19,16 @@ export const GAS_LIMIT_HARDHAT: BigNumber = bn("1e8");
...
+export const H_TOKEN_SYMBOL: string = `hUSDC${H_TOKEN_EXPIRATION_DATE_SHORT}`;
```
[the remark this answers is not in the document]
Comment (scorpion9979) [reply helpers/constants.ts]: done

File: helpers/numbers.ts [hunk 8]
```diff
@@ -18,3 +18,9 @@ export function price(x: string): BigNumber {
 export function usdc(x: string): BigNumber {
   return fp(x, 6);
 }
+
+export function tokenWithNDecimalsPrecisionScalar(n: number): BigNumber {
```
[the remark this answers is not in the document]
Comment (scorpion9979) [reply helpers/numbers.ts]: done

File: helpers/numbers.ts (hunk 1 above)
Comment (scorpion9979) [reply helpers/numbers.ts]: done

File: helpers/numbers.ts (hunk 2 above)
Comment (scorpion9979) [reply helpers/numbers.ts]: done

File: test/unit/chainlinkOperator/ChainlinkOperator.behavior.ts (hunk 3 above)
Comment (scorpion9979) [reply test/unit/chainlinkOperator/ChainlinkOperator.behavior.ts]: done

**Review by paulrberg — APPROVED** [review]

**Pull request merged**